src/components/ButtonSystem/BaseButton.tsx

```diff
@@ -0,0 +1,85 @@
+/**
+ * Not meant to be edited
+ *
+ * This is the base button for all button variations (internal/external links, hash links, non-link buttons)
+ *
+ * Styles/variations are added in MainButtons.style.tsx
+ */
+
+import React, { ComponentProps, DetailedHTMLProps } from 'react';
+import {
+  Link as RouterLink,
+  LinkProps as RouterLinkProps,
+} from 'react-router-dom';
+import { HashLink, HashLinkProps } from 'react-router-hash-link';
+import { assert } from 'common/utils';
+import { BaseMuiButton } from './BaseButton.style';
+import { EventAction, EventCategory, trackEvent } from 'components/Analytics';
+
+type AnchorLinkType = DetailedHTMLProps<
+  React.AnchorHTMLAttributes<HTMLAnchorElement>,
+  HTMLAnchorElement
+>;
+
+type StyledButtonProps = ComponentProps<typeof BaseMuiButton>;
+
+type LinkProps = AnchorLinkType | RouterLinkProps | HashLinkProps;
+
+interface TrackingProps {
+  trackingCategory: EventCategory;
+  trackingAction?: EventAction;
+  trackingLabel: string;
+}
+
+export type LinkButtonProps = LinkProps & TrackingProps & StyledButtonProps;
+
+const BaseButton: React.FC<LinkButtonProps> = props => {
```

Comment:
Just writing here so we don't forget, let's make sure that the `trackingCategory` and `trackingLabel` props are mandatory for both buttons and links. We can make `trackingAction` non-mandatory since it's kind of implied by the type of element ( `NAVIGATION` for links and `CLICK` for regular buttons)

Reply:
all set